internal/core/command/device.go

```diff
@@ -94,24 +120,33 @@ func executeCommandByName(
 		return "", err
 	}
 
-	return executeCommandByDevice(ctx, d, command, body, queryParams, isPutCommand, lc)
+	return executeCommandByDevice(ctx, d, command, body, lc, r, httpCaller)
 }
 
 func executeCommandByDevice(
 	ctx context.Context,
 	device contract.Device,
 	command contract.Command,
 	body string,
-	queryParams string,
-	isPutCommand bool,
-	lc logger.LoggingClient) (string, error) {
+	lc logger.LoggingClient,
+	r *http.Request,
+	httpCaller internal.HttpCaller) (string, error) {
 
+	var method string
 	var ex Executor
 	var err error
-	if isPutCommand {
-		ex, err = NewPutCommand(device, command, body, ctx, &http.Client{}, lc)
-	} else {
-		ex, err = NewGetCommand(device, command, queryParams, ctx, &http.Client{}, lc)
+
+	if r == nil {
+		return "", errors.NewErrRequestParameterSupplied("")
```

Comment:
I brought this up in a previous iteration and I think it got lost in all the comments, but I think we should change this to fit the situation better. The `ErrRequestParamenterSupplied` is never constructed with any arguments and therefore the resulting error message is not helpful. Can we either update the error struct to have a message that does not refer to the parameters in question(since they are never supplied anyway) or pass in the the parameters you are expecting so the error message is useful. Furthermore, I do not think this error fits this situation anyway since you are looking for the HTTP method and not any parameters.